docs/design-tech/sync.md

```diff
@@ -1,89 +1,263 @@
-# Syncing within core
+# Sync v3
+## Overview
+Sync v3 is the latest iteration of our logic that syncs users' file trees across devices. It is designed under the following key constraints:
+* The state of a user's file tree must satisfy certain invariants at all times, on all devices (including the server), even in the event of process interruptions and failures.
+* Users' file contents and names must not be readable by the server, but users can share files and folders with other users.
+* Users must be able to view and edit files offline, then sync frictionlessly, even with intervening edits made to files on other devices.
+* The performance of the system must not suffer as a result of typical usage, which presents a number of challenges worth discussion
 
-Understanding the [data model](data_model.md) is a prerequisite to understanding this document.
+These design constraints motivate the following high-level design, which will be flushed out in more detail after a further discussion of the constraints:
+* The client maintains two versions of the file tree: one for the current local version (`local`), and one for the last version known to be agreed upon by the client and server (`base`).
```

Comment:
A potential useful disambiguation: `local` is empty when there are no changes locally. We don't just take up 2x the storage locally.

Resolve if you disagree.